web/src/unread_ops.ts

```diff
@@ -256,31 +258,71 @@ export function mark_as_unread_from_here(
     }
     message_lists.current.prevent_reading();
 
-    // If we have already fully fetched the current view, we can
-    // send the server the set of IDs to update, rather than
-    // updating on the basis of the narrow.
-    let message_ids_to_update;
-    if (message_lists.current.data.fetch_status.has_found_newest()) {
-        message_ids_to_update = message_lists.current
-            .all_messages()
-            .filter(
-                (msg) =>
-                    (include_anchor && msg.id >= message_id) ||
-                    (!include_anchor && msg.id > message_id),
-            )
-            .map((msg) => msg.id);
+    const rounding_factor = 25;
+    const threshold_fetched = 50;
+    const treshold_unfetched = 10;
+    const is_all_fetched = message_lists.current.data.fetch_status.has_found_newest();
+    const is_in_interleaved_view = !narrow_state.narrowed_by_reply();
+    let message_ids_to_update: number[] | undefined;
+
+    function do_mark_unread(): void {
+        // If we have already fully fetched the current view, we can
+        // send the server the set of IDs to update, rather than
+        // updating on the basis of the narrow.
+        if (is_all_fetched && message_ids_to_update!.length < 200) {
```

Comment:
Use `assert` from `minimalistic-assert`, or maybe you can do `message_ids_to_update !== undefined` instead of using `is_all_fetched`